### ISSUE TITLE
`valueOf` causes NPE on Android for sealed classes

### ISSUE THREAD

**Issue by vitorhugods** [mockative/mockative#87]
Since Mockative 2.x, I'm unable to run some tests on Android, as Mockative causes an exception when using `ProxyFactory$create`.

```
java.lang.NullPointerException: Attempt to read from null array
at javassist.util.proxy.SecurityActions.getCallerClass(SecurityActions.java:54)
at javassist.util.proxy.DefineClassHelper$Java7.defineClass(DefineClassHelper.java:174)
at javassist.util.proxy.DefineClassHelper.toClass(DefineClassHelper.java:260)
at javassist.util.proxy.FactoryHelper.toClass(FactoryHelper.java:154)
at javassist.util.proxy.ProxyFactory.createClass3(ProxyFactory.java:640)
at javassist.util.proxy.ProxyFactory.createClass2(ProxyFactory.java:624)
at javassist.util.proxy.ProxyFactory.createClass1(ProxyFactory.java:560)
at javassist.util.proxy.ProxyFactory.createClass(ProxyFactory.java:481)
at javassist.util.proxy.ProxyFactory.create(ProxyFactory.java:828)
at io.mockative.fake.ValueOfJVM.makeValueOf(ValueOf.kt:55)
at io.mockative.fake.ValueOfJVM.makeValueOf(ValueOf.kt:31)
at io.mockative.fake.ValueOfKt.valueOf(ValueOf.kt:84)
at com.wire.kalium.logic.data.auth.login.LoginRepositoryTest$Arrangement$withLoginReturning$2$1.invokeSuspend(LoginRepositoryTest.kt:134)
at com.wire.kalium.logic.data.auth.login.LoginRepositoryTest$Arrangement$withLoginReturning$2$1.invoke(Unknown Source:8)
at com.wire.kalium.logic.data.auth.login.LoginRepositoryTest$Arrangement$withLoginReturning$2$1.invoke(Unknown Source:2)
at io.mockative.Mockable$Companion.record(Mockable.kt:358)
at io.mockative.EveryKt.coEvery(Every.kt:28)
```

The code in question looks like this:
```kotlin
coEvery {
    loginApi.login(any(), any())
}.returns(response)
```

I am not 100% sure, but it seems related to the `any()` trying to instantiate a placeholder `sealed` class and that seems to fail on Android, when [calling `ProxyFactory.create`](https://github.com/mockative/mockative/blob/76853d4c2a69d8c707dfd6cfdb65ea21e0922d14/mockative/src/jvmMain/kotlin/io/mockative/fake/ValueOf.kt#L55).

I want to eventually dive deeper into this.

**Comment by vitorhugods** [comment]
It seems to really be related to `sealed` classes.

When running [`isSealed(clazz)`](https://github.com/mockative/mockative/blob/76853d4c2a69d8c707dfd6cfdb65ea21e0922d14/mockative/src/jvmMain/kotlin/io/mockative/fake/ValueOf.kt#L21) on Android, it always returns `false`, so it goes into the [`isAbstract`](https://github.com/mockative/mockative/blob/76853d4c2a69d8c707dfd6cfdb65ea21e0922d14/mockative/src/jvmMain/kotlin/io/mockative/fake/ValueOf.kt#L46) case, trying to create an instance of this sealed class.

I'm unsure on how to proceed. Maybe dealing with `KClass` instead of `Class` would be best in some places.

**Comment by vitorhugods** [comment]
Fixed in #92
